### ISSUE TITLE
MariaDB dialect does not seem to support partitioning by LIST with VALUES IN clause

### ISSUE THREAD

**Issue by rtista** [sqlfluff/sqlfluff#7648]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The sqlfluff execution is failing parsing with for this simple table, and from what I've gathered it seems to be a problem with the VALUES IN clause within a LIST type partitioning.

```sql
CREATE TABLE IF NOT EXISTS `tablename`
(
    `part_id` INT(11) NOT NULL,
    `uid` CHAR(10) NOT NULL,
    PRIMARY KEY (`part_id`, `uid`)
) ENGINE = Aria PARTITION BY LIST (`part_id`) (
    PARTITION `p0` VALUES IN (0)
);
`

I've tested with PARTITION BY RANGE with VALUES LESS THAN clause and it works, so I believe its something to do with the expected value after the VALUES IN clause.

### Expected Behaviour

For the table specification to be considered valid as it is expected.

### Observed Behaviour

$ sqlfluff lint test.sql --dialect mariadb
== [test.sql] FAIL                                                                                                                                                                                                   
L:   6 | P:  47 |  PRS | Line 6, Position 47: Found unparsable section: '(\n   
                       | PARTITION `p0` VALUES IN (0)\n);'
WARNING: Parsing errors found and dialect is set to 'mariadb'. Have you configured your dialect correctly?
All Finished 📜 🎉!


### How to reproduce

Use the provided SQL and sqlfluff for mariadb or mysql dialect without any configurations.

### Dialect

MariaDB / MySQL

### Version

4.0.4

### Configuration

No configuration.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
